### PR TITLE
Check if tables are empty before seeding

### PR DESCRIPTION
**Pull request by StefanGajic**
https://trello.com/c/3MkKDlN7/129-check-whether-all-tables-are-empty-before-seeding

### CODE REVIEW

File: seed/main.go [hunk 1]
```diff
@@ -47,10 +48,19 @@ func main() {
...
-		if err := store.CreateBook(book); err != nil {
-			panic(err)
+
+	db := store.GetDB()
+
+	bookList := []model.Book{}
+	var num int
+	db.Model(&bookList).Count(&num)
+
+	if num < 0 {
```
Comment (bokimilinkovic) [inline seed/main.go]: ==

File: seed/main.go [hunk 2]
```diff
@@ -47,10 +48,19 @@ func main() {
...
-		if err := store.CreateBook(book); err != nil {
-			panic(err)
+
+	db := store.GetDB()
+
+	bookList := []model.Book{}
+	var num int
+	db.Model(&bookList).Count(&num)
+
+	if num == 0 {
```
Comment (sfqi) [inline seed/main.go]: Use an early return (`if num != 0`)

Comment (sfqi) [reply seed/main.go]: Also `num` isn't great as a name. Maybe write a function that will check whether the table is empty - for readability

File: seed/main.go [hunk 3]
```diff
@@ -32,6 +34,16 @@ var books = []*model.Book{
 	},
 }
 
+func areTablesEmpty(db *gorm.DB) bool {
+	var num int
+	db.Model([]*model.Book{}).Count(&num)
+
+	if num > 1 {
+		return true
```
Comment (sfqi) [inline seed/main.go]: this can't be right

File: seed/main.go [hunk 4]
```diff
@@ -47,10 +59,13 @@ func main() {
...
+
+	if !areTablesEmpty(store.GetDB()) {
+		for _, book := range books {
+			if err := store.CreateBook(book); err != nil {
+				panic(err)
+			}
 		}
 	}
-
+	fmt.Println("Tables are not empty")
```
Comment (sfqi) [inline seed/main.go]: This doesn't really make sense. We should panic if the tables are empty and that's it

File: seed/main.go [hunk 5]
```diff
@@ -47,10 +58,13 @@ func main() {
...
+
+	if !areTablesEmpty(store.GetDB()) {
+		for _, book := range books {
+			if err := store.CreateBook(book); err != nil {
+				panic(err)
+			}
 		}
 	}
-
+	panic("Tables are not empty")
```
Comment (sfqi) [inline seed/main.go]: It would be nice to know which table isn't empty

File: seed/main.go (hunk 3 above)
Comment (sfqi) [reply seed/main.go]: What's the status on this?

Comment (StefanGajic) [reply seed/main.go]: I have changed it to 	```
if num > 0 {
		return true
```
I am not sure if you saw it

Comment (sfqi) [reply seed/main.go]: So what this function says it that if there's more than 0 rows in the table, the table is empty? Am I misreading anything?

File: seed/main.go [hunk 6]
```diff
@@ -32,6 +33,16 @@ var books = []*model.Book{
 	},
 }
 
+func areTablesEmpty(db *gorm.DB) bool {
+	var num int
+	db.Model([]*model.Book{}).Count(&num)
```
Comment (sfqi) [inline seed/main.go]: looking at this example from gorm:
```
db.Model(&User{}).Where("name = ?", "jinzhu").Count(&count)
```

I'm guessing we don't need a slice here

**Comment by sfqi** [comment]
Also, could you merge https://github.com/sfqi/library/pull/91 in here and do the same for loans?

File: seed/main.go [hunk 7]
```diff
@@ -1,37 +1,74 @@
...
+		return err
+	}
+
+	if bookNum != 0 {
+		return errors.New("Table book is not empty!")
+	}
+	if loanNum != 0 {
+		return errors.New("Table loan is not empty!")
+	}
+	return nil
```
Comment (sfqi) [inline seed/main.go]: you might be noticing that the code is a bit repetitive here (imagine we had 12 tables)
can you refactor it?

also, why `num`? `count`!

Comment (sfqi) [reply seed/main.go]: 1. You  declare `count`, it's automatically set to zero, and then you switch based on the value of count? It's always zero. (so are `bookCount` and `loanCount`) 
2. You didn't really achieve much in the way of having less repetition. Again, imagine there are 12 tables - you'd end up with a massive function
3. Hint: you need only one `count` var

**Pull request merged**